Extension/src/LanguageServer/extension.ts

```diff
@@ -343,6 +344,47 @@ function onToggleSquiggles(): void {
     settings.toggleSetting("errorSquiggles", "Enabled", "Disabled");
 }
 
+function onToggleSnippets(): void {
+    onActivationEvent();
+
+    // This will apply to all clients as it's a global toggle. It will require a reload.
+    const snippetsCatName: string  = "Snippets";
+    const snippetsNodeName: string = "snippets";
+    let newPackageJson: any = util.getRawPackageJson();
+
+    if (newPackageJson.categories.findIndex(cat => cat === snippetsCatName) === -1) {
+        // Add the Snippet category and snippets node. 
+
+        newPackageJson.categories.push(snippetsCatName);
+        newPackageJson.contributes[snippetsNodeName] = [{"language": "cpp", "path": "./cpp_snippets.json"}, {"language": "c", "path": "./cpp_snippets.json"}];
+
+        fs.writeFileSync(util.getPackageJsonPath(), util.stringifyPackageJson(newPackageJson));
+        showReloadPrompt("Reload Window to finish enabling C++ snippets");
+    }
+    else {
+        // Remove the category and snippets node.
+
+        let ndxCat = newPackageJson.categories.indexOf(snippetsCatName);
+        if (ndxCat != -1)
+            newPackageJson.categories.splice(ndxCat, 1);
+
+        delete newPackageJson.contributes[snippetsNodeName];
+
+        fs.writeFileSync(util.getPackageJsonPath(), util.stringifyPackageJson(newPackageJson));
```

Comment:
nit: this could be async with the `showReloadPrompt` in a `.then` continuation function